R/reporter-stop.R

```diff
@@ -33,9 +33,15 @@ StopReporter <- R6::R6Class("StopReporter",
       self$failures$initialize()
 
       messages <- vapply(failures, format, character(1))
-      messages <- paste0("* ", messages, collapse = "\n")
+      locations <- vapply(failures, exp_location, character(1))
+      messages <- paste0("* ", locations, ": ", messages, collapse = "\n")
+      message <- paste_line(
+        paste0("Test failed: '", test, "'"),
+        !!!messages
+      )
 
-      stop("Test failed: '", test, "'\n", messages, call. = FALSE)
+      cat(message, "\n", file = stderr())
+      invokeRestart("abort")
```

Comment:
Doesn't this immediately jump to the top level? Is that too aggressive?

Reply:
I don't know the usage of the `Stop` reporter but for interactive applications this seemed appropriate. We can also throw a special class that we'd rethrow from `handle_fatal()`, this way it could be caught again from outside testthat.

Reply:
I think throwing a special class might be safer. Otherwise how do we test it?

Reply:
We can add another abort restart on the stack to test it. No worries for a special class.

Reply:
`with_reporter()` now installs a `testthat_abort_reporter` restart that is invoked by the stop reporter.